File: 基礎題庫/Contents/b367/b367.py
import sys

for T in sys.stdin:
    try:
        T = int(T.strip())
    except:
        continue
    for i in range(T):
        N, M = map(int, input().split())

        mat = []
        for j in range(N):
            mat += input().split()
        
        mat_rev = [e for e in reversed(mat)]
        if mat == mat_rev:
            print("go forward")
        else:
            print("keep defending")

File: 基礎題庫/Contents/b367/b367.c
#include<stdio.h>

int main(void)
{
	int T, N, M;
	int mat[122];
	
	while(scanf("%d", &T) != EOF)
	{
		for(int i = 0; i < T; i++)
		{
			scanf("%d %d", &N, &M);
			
			for(int j = 0; j < N * M; j++)
				scanf("%d", &mat[j]);
			
			int head = 0;
			int tail = M * N - 1;
			while(mat[head] == mat[tail])
			{
				if(head > tail)
					break;
				
				head++;
				tail--;
			}
			
			printf("%s\n", head < tail ? "keep defending" : "go forward");
		}
	}
	
	return 0;
}
